src/gtk/srchctrl.cpp

```diff
@@ -0,0 +1,429 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/srchctrl.cpp
+// Purpose:     wxSearchCtrl implementation - native
+// Author:      Kettab Ali
+// Created:     2019-12-23
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#if wxUSE_SEARCHCTRL
+
+#include "wx/srchctrl.h"
+
+#ifndef WX_PRECOMP
+    #include "wx/menu.h"
+#endif //WX_PRECOMP
+
+#include "wx/utils.h"
+#include "wx/gtk/private.h"
+#include "wx/gtk/private/gtk3-compat.h"
+
+
+#if GTK_CHECK_VERSION(3,6,0)
+    // GtkSearchEntry is available only for GTK+ >= 3.6
+    #define wxHAS_GTK_SEARCH_ENTRY 1
```

Comment:
For consistency, this one should be just defined (and not defined at all in the `#else` branch).